### DISCUSSION TITLE
Run an action after a non zero return code in a csharp script

### DISCUSSION THREAD

**Discussion by ChristopherL-STCU** [alirezanet/Husky.Net#26] [Q&A] [answered]
I trying to use the commit message linter example to prevent invalid commit messages. It seems if the csharp script returns with a code of `1` then all execution is halted. But I'd like to perform a soft reset of the commit if there is a non zero return code. Is that possible?

I'd like to do something like this:
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

message=$(git log -n1 --pretty="%s")
dotnet husky run --name "commit-message-linter" --args "$message"
if [ $? -eq 1 ]
then
    git reset --soft HEAD^
fi
```

Is something like this possible? Or some other way to accomplish what I'm trying to do?

**Comment by alirezanet** [comment, marked as answer]
Did you try to use the `exec` command directly in the hook? 
e.g
``` shell
dotnet husky exec "CsharpScriptPath.csx" --args "$message"
```
I think you don't need the husky `run` because it will stop if receives any non-zero exit code.  but `exec` should work fine.

**Reply by ChristopherL-STCU** [reply]
Thank you for the response. That doesn't seem to work either. If I change the hook to:
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

message=$(git log -n1 --pretty="%s")
dotnet husky exec ./.husky/csx/commit-message-linter.csx --args "$message"
echo
echo Great work! 🥂
echo
```
The content after the `exec` is only echoed when the return code is 0. When the return code is 1 then none of the echo content is displayed.

**Reply by alirezanet** [reply]
You are right, I have another question why are you trying to get the last committed message  `message=$(git log -n1 --pretty="%s")`  instead of preventing the commit in the first place (using `commit-msg` hook)? 

I'll try to find a solution to get the csx script output back. till then please explain the use-case a little bit, I still don't understand what are you trying to achieve? Although you can move the entire logic to the csx file. the whole point of it is to use c# code instead of sh scripts.

**Reply by alirezanet** [reply]
I managed to get the exit code by disabling husky.sh script which is fine in this case.
e.g `commit-msg` hook
``` shell
#!/bin/sh
# prevent breaking the execuition by disabling hook handler
#. "$(dirname "$0")/_/husky.sh"

message=$(git log -n1 --pretty="%s")
dotnet husky exec .husky/linter.csx --args "$message"

if [ $? -eq 0 ]
then
  echo "Success: "
  exit 0
else
  echo "Failure: " >&2
  exit 1
fi
```

e.g `linter.csx`
``` csharp
using System.Text.RegularExpressions;

private var pattern = @"^(?=.{1,90}$)(?:build|feat|ci|chore|docs|fix|perf|refactor|revert|style|test)(?:\(.+\))*(?::).{4,}(?:#\d+)*(?<![\.\s])$";

var filePath = Args[0];
private var msg = "";

// file mode
if (File.Exists(filePath))
   msg = File.ReadAllLines(filePath)[0];
else // msg-mode
   msg = Args[0];

if (Regex.IsMatch(msg, pattern))
   return 0;

Console.ForegroundColor = ConsoleColor.Red;
Console.WriteLine("Invalid commit message");

return 1;
```
this works, but still `git log -n1 --pretty="%s"` doesn't give you the current commit message unless you use it for example in the `post-commit` hook which is weird

**Reply by ChristopherL-STCU** [reply]
Mainly because I took a very winding path and ended up here. So not a legitimate reason to do it like that, at least for this. Thank you for all of your help. 👏
